Add page around PR checks to contribution guidelines. (#4006)
open-telemetry/opentelemetry.io: content/en/docs/contributing/style-guide.md

The following sections contain guidance that is specific to the OpenTelemetry
project.

{{% alert title="Note" color="primary" %}}

Many requirements of our style guide can be enforced by running automation:

> before providing a

Reviewer: ```suggestion
before submitting a
```

[pull request](https://docs.github.com/en/get-started/learning-about-github/github-glossary#pull-request)

> (PR) run `npm run fix:all` on your local machine and commit the changes.

Author: ```suggestion
(PR), run `npm run fix:all` on your local machine and commit the changes.
```

> (PR) run `npm run test-and-fix` on your local machine and commit the changes.

Author: ```suggestion
(PR) run `npm run fix:all` on your local machine and commit the changes.
```

> (PR) run `npm run test-and-fix` on your local machine and commit the changes.

Reviewer: ```suggestion
(PR), run `npm run test-and-fix` on your local machine and commit the changes.
```

Author: fixed

See also the [Glossary](/docs/concepts/glossary/) for a list of OpenTelemetry
terms and their definition.

> Run `npm run check:text` to verify that all terms and words are written
> properly.

Author: ```suggestion
Run `npm run check:text` to verify that all terms and words are written
properly.

Run `npm run check:text -- --fix` to fix terms and words that are not written properly.
```

See also the [Glossary](/docs/concepts/glossary/) for a list of OpenTelemetry
terms and their definition.

Run `npm run check:text` to verify that all terms and words are written
properly.

## Markdown standards

To enforce standards and consistency for Markdown files, all files should follow
certain rules, enforced by
[markdownlint](https://github.com/DavidAnson/markdownlint). For a full list,
check the
[`.markdownlint.json`](https://github.com/open-telemetry/opentelemetry.io/blob/main/.markdownlint.json)
file.

Run `npm run check:markdown` to verify that all files follow the standard.

> Run `npm run fix:format` to fix markdown related formatting issues.

Reviewer 2: ```suggestion
Run `npm run fix:format` to fix Markdown related formatting issues.
```

> Run `npm run fix:format` to fix Markdown related formatting issues.

Reviewer: ```suggestion
Run `npm run fix:markdown` to fix Markdown related formatting issues.
```

## Spell checking

Use [CSpell](https://github.com/streetsidesoftware/cspell) to make sure that all
your text is spelled correctly. For a list of words that are specific to the

> OpenTelemetry website, check the

Reviewer: ```suggestion
OpenTelemetry website, see the
```

[`.cspell.yml`](https://github.com/open-telemetry/opentelemetry.io/blob/main/.cspell.yml)
file.

Run `npm run check:spelling` to verify that all your words are spelled
correctly. If `cspell` indicates an `Unknown word` error, verify if you wrote
that word correctly. If so, add this word to the `cSpell:ignore` section at the

> top of your file. If no such section exists, you can add it to the frontmatter

Author: ```suggestion
top of your file. If no such section exists, you can add it to the front matter
```

> of a Markdown file or as a comment at the top of any other file:

Reviewer: ```suggestion
of a Markdown file:
```

```markdown
---
title: PageTitle
cSpell:ignore: <word>
---
```

> 

Reviewer: ```suggestion
For any other file, add `cSpell:ignore <word>` in a comment line appropriate for the file's context. For a [registry](/ecosystem/registry/) entry YAML file file, it might look like this:

```

```yaml
# cSpell:ignore <word>
title: registryEntryTitle
```

{{% alert title="Note" color="warning" %}}

> If you have multiple words in the cSpell ignore list, please run

Reviewer 2: ```suggestion
If you have multiple words in the cSpell ignore list, run
```

> {{% alert title="Note" color="warning" %}}
> 
> If you have multiple words in the cSpell ignore list, run
> `npm run fix:dict` to make sure that those words are in alphabetical order.
> 
> {{% /alert %}}

Reviewer: ```suggestion
Website tooling normalizes page-specific dictionaries (that is, the `cSpell:ignore` word lists), by removing duplicate words, deleting words in the global word list, and sorting words. To normalize page-specific dictionaries, run `npm run fix:dict`.
```

## File format

To enforce a certain standard on how files are structured, all files should be
formatted by [prettier](https://prettier.io). Run `npm fix:format` before
submitting a PR, or run it afterwards and push an additional commit.

## File names

> All file names should should be in

Reviewer 2: ```suggestion
All file names should be in
```

Reviewer 2: I'm tempted to replace all "should" with "must" in this page.

file.

Run `npm run check:markdown` to verify that all files follow the standard.

Run `npm run fix:markdown` to fix Markdown related formatting issues.

## Spell checking

Use [CSpell](https://github.com/streetsidesoftware/cspell) to make sure that all
your text is spelled correctly. For a list of words that are specific to the
OpenTelemetry website, see the
[`.cspell.yml`](https://github.com/open-telemetry/opentelemetry.io/blob/main/.cspell.yml)
file.

Run `npm run check:spelling` to verify that all your words are spelled
correctly. If `cspell` indicates an `Unknown word` error, verify if you wrote
that word correctly. If so, add this word to the `cSpell:ignore` section at the
top of your file. If no such section exists, you can add it to the front matter
of a Markdown file:

```markdown
---
title: PageTitle
cSpell:ignore: <word>
---
```
For any other file, add `cSpell:ignore <word>` in a comment line appropriate for the file's context. For a [registry](/ecosystem/registry/) entry YAML file file, it might look like this:

```yaml
# cSpell:ignore <word>
title: registryEntryTitle
```

> {{% alert title="Note" color="warning" %}}
> 
> If you have multiple words in the cSpell ignore list, run `npm run fix:dict` to
> make sure that those words are in alphabetical order.
> 
> {{% /alert %}}

Author: ```suggestion
Website tooling normalizes page-specific dictionaries (that is, the `cSpell:ignore` word lists), by removing duplicate words, deleting words in the global word list, and sorting words. To normalize page-specific dictionaries, run `npm run fix:dict`.
```
